qiskit/utils/run_circuits.py

```diff
@@ -146,7 +146,7 @@ def _safe_submit_qobj(
     skip_qobj_validation: bool,
 ) -> Tuple[BaseJob, str]:
     # assure get job ids
-    while True:
+    for attempt in range(10):
```

Comment:
Instead of doing this a fixed number of times do you think it would make sense to let the retry count be configurable by the user? I feel like this should be configurable with some thing like `while run_count < retry_limit` then people can specify how many times to retry before giving up (or specify -1 to get the current infinite behavior if they wanted that for some reason).

Reply:
Yes, that could be provided as a `max_tries` parameter to a `job` instance whereas the current count was just chosen at random. I think a parameter like `max_tries` in a `job` instance would fit the role here. 
Also, one small caveat here is that I am not sure if we can actually say that each try would experience the same **latency** in the network. If the latency varies, then some tries are **more expensive** than others although they still are a single try( not perfectly sure about whether this affects how big the number of tries are)

Reply:
Is there any update on this as I have the commit with the requested changes ready to be pushed? Thanks!

Reply:
In these cases, if you completely agree with the implementation suggested, you can just push the change and we'll review it.

You don't have to worry about latency varying with different tries - this isn't specific to networking anyway; any program running on a modern computer has fluctuating speeds due to the OS scheduling other things.  The main reason for limiting the number of tries is to prevent infinite loops, rather than trying to absolutely fix the time spent waiting for something to succeed.  There's always the possibility of having a time-based timeout option instead, but these are harder to do correctly - if you wait for the loop condition check to test the time, a slow-running network will cause you to overshoot your timeout, so you need to have some asynchronicity.  `max_tries` is a common pattern, and nice and easy to implement for both us and users.

Reply:
The original thinking when this was done was the loop would break if there was an unrecoverable error otherwise it treated errors as recoverable and would continue. Of course if something truly is unrecoverable then it would require end-user action to terminate their program; but it has been a while since that code was done and things have changed. What we did not want to happen was some temporary remote system shutdown/failure, some maintenance etc end up causing a long running iterative/variational algorithm to terminate as these, especially when using VQE, could take many hours to run with many jobs submitted as the algorithm proceeds. So my concern would be that if this ends due to errors around some temporary unavailability due to some loop limit being quickly reached. User configurability of any retry limit ought to be there at a minimum - i.e. if its recoverable why only try 10 times - of course its a guard against it actually being non-recoverable but unlike and end user taking manual action to Ctrl-C the program if it seems to be stuck this if done incorrectly will abend things when the algorithm otherwise would have continued to completion once the system issue was rectified.

Reply:
Okay, if that is the case, where would it be optimal to place a `max_tries` parameter for the user? My initial thought was that `execute` and `backend.run` methods would need to be updated and the relevant functions inside the `run_circuits` file. Thanks!